code/Base/GamemodeSystem/Entities/BaseGamemode.TargetEntities.cs

```diff
@@ -0,0 +1,44 @@
+namespace Sports;
+
+public partial class BaseGamemode
+{
+	public List<Entity> Entities { get; protected set; } = new();
+
+	public IEnumerable<Entity> GetEntitiesWithTag( string tag )
+	{
+		return Entities.Where( x => x.Tags.Has( tag ) );
+	}
+	public IEnumerable<T> GetEntitiesOfType<T>( string tag = null )
+	{
+		if ( tag is not null )
+		{
+			return Entities.OfType<T>();
+		}
+		else
+		{
+			return GetEntitiesWithTag( tag )
+				.OfType<T>();
+		}
+	}
+
+	[Event.Entity.PostSpawn]
+	protected virtual void GatherTargetEntities()
+	{
+		var ents = FindAllByName( Name );
```

Comment:
for the sake of I/O targetnames we should probably use a property on GamemodePointEntity that links them to the gamemode instead, similar to how we do it with the orbs.